crates/router/src/core/utils.rs

```diff
@@ -5,19 +5,142 @@ use common_utils::errors::CustomResult;
+            let (business_country, business_label) = helpers::get_business_details(
+                payout_create.business_country,
+                payout_create.business_label.as_ref(),
+                merchant_account,
+            )?;
+
+            let connector_label =
+                helpers::get_connector_label(business_country, &business_label, None, connector_id);
+
+            let merchant_connector_account = helpers::get_merchant_connector_account(
+                state,
+                merchant_account.merchant_id.as_str(),
+                &connector_label,
+                None,
+            )
+            .await?;
+            Ok(merchant_connector_account)
+        }
+    }
+}
+
+#[instrument(skip_all)]
+#[allow(clippy::too_many_arguments)]
+pub async fn construct_payout_router_data<'a, F>(
+    state: &'a AppState,
+    connector_id: &str,
+    merchant_account: &storage::MerchantAccount,
+    _request: &api_models::payouts::PayoutRequest,
+    payout_data: &mut PayoutData,
+    payout_method_data: &api::PayoutMethodData,
+) -> RouterResult<types::PayoutsRouterData<F>> {
+    let merchant_connector_account =
+        get_mca_for_payout(state, connector_id, merchant_account, payout_data).await?;
+    payout_data.merchant_connector_account = Some(merchant_connector_account.clone());
+    let connector_auth_type: types::ConnectorAuthType = merchant_connector_account
+        .get_connector_account_details()
+        .parse_value("ConnectorAuthType")
+        .change_context(errors::ApiErrorResponse::InternalServerError)?;
+
+    let billing = payout_data.billing_address.to_owned();
+
+    let address = PaymentAddress {
+        shipping: None,
+        billing: billing.map(|a| {
+            let phone_details = api_models::payments::PhoneDetails {
+                number: a.phone_number.to_owned(),
+                country_code: a.country_code.to_owned(),
+            };
+            let address_details = api_models::payments::AddressDetails {
+                city: a.city.to_owned(),
+                country: a.country.to_owned(),
+                line1: a.line1.to_owned(),
+                line2: a.line2.to_owned(),
+                line3: a.line3.to_owned(),
+                zip: a.zip.to_owned(),
+                first_name: a.first_name.to_owned(),
+                last_name: a.last_name.to_owned(),
+                state: a.state,
+            };
+
+            api_models::payments::Address {
+                phone: Some(phone_details),
+                address: Some(address_details),
+            }
+        }),
+    };
+
+    let payouts = &payout_data.payouts;
+    let payout_create = &payout_data.payout_create;
+
+    let router_data = types::RouterData {
+        flow: PhantomData,
+        merchant_id: merchant_account.merchant_id.to_owned(),
+        customer_id: None,
+        connector_customer: None,
+        connector: connector_id.to_string(),
+        payment_id: "".to_string(),                      //FIXME
```

Comment:
will it be handled in this PR if not add a issue and link it here

Reply:
We don't need `payment_id` for payouts use-case, will remove the comment